File: windows.ui.xaml/code/DOandDP/csharp/Class1.cs
using System;
using System.Text;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;
using Windows.UI.Xaml.Input;
using Windows.UI.Xaml.Media;
using Windows.UI.Xaml.Media.Animation;
using Windows.UI.Xaml.Interop;

namespace DOandDP
{
//<SnippetDOMain>
    public abstract class AquariumServices : DependencyObject
    {
        public enum Buoyancy {Floats,Sinks,Drifts}

        public static readonly DependencyProperty BuoyancyProperty = DependencyProperty.RegisterAttached(
          "Buoyancy",
          typeof(Buoyancy),
          typeof(AquariumServices),
          new PropertyMetadata(Buoyancy.Floats)
        );
        public static void SetBuoyancy(DependencyObject element, Buoyancy value)
        {
            element.SetValue(BuoyancyProperty, value);
        }
        public static Buoyancy GetBuoyancy(DependencyObject element)
        {
            return (Buoyancy)element.GetValue(BuoyancyProperty);
        }
    }
    //</SnippetDOMain>

    //<SnippetDOSimpleDP>
    public class Fish : Control
    {
        public static readonly DependencyProperty SpeciesProperty =
        DependencyProperty.Register(
        "Species",
        typeof(String),
        typeof(Fish), null
        );
        public string Species
        {
            get { return (string)GetValue(SpeciesProperty); }
            set { SetValue(SpeciesProperty, (string)value); }
        }
    }
    //</SnippetDOSimpleDP>
    public static class DependencyPropertyUtilities
    {
        //<SnippetDOCheckClear>
        public static bool ClearSetProperty(DependencyObject targetObject, DependencyProperty targetDP)
        {
            if (targetObject == null || targetDP == null)
            {
                throw new ArgumentNullException();
            }
            object localValue = targetObject.ReadLocalValue(targetDP);
            if (localValue == DependencyProperty.UnsetValue)
            {
                return false;
            }
            else
            {
                targetObject.ClearValue(targetDP);
                return true;
            }
        }
        //</SnippetDOCheckClear>
        //<SnippetDPGetMetadata>
        public static object GetDefaultValueForFrameworkDP(DependencyProperty dpIdentifier)
        {
            PropertyMetadata metadataInfo = dpIdentifier.GetMetadata(typeof(FrameworkElement));
            return metadataInfo.DefaultValue;
        }
        //</SnippetDPGetMetadata>
    }
}

File: windows.ui.xaml/code/DOandDP/vbnet/Class1.vb
Public Class Class1

End Class

'<SnippetDOMain>
Public Class AquariumServices
    Inherits DependencyObject
    Public Enum Buoyancy
        Floats
        Sinks
        Drifts
    End Enum

    Public Shared ReadOnly BuoyancyProperty As DependencyProperty = _
          DependencyProperty.RegisterAttached(
          "Buoyancy", _
          GetType(Buoyancy), _
          GetType(AquariumServices), _
          New PropertyMetadata(Buoyancy.Floats))


    Public Sub SetBuoyancy(element As DependencyObject, value As Buoyancy)
        element.SetValue(BuoyancyProperty, value)
    End Sub
    Public Function GetBuoyancy(element As DependencyObject) As Buoyancy
        GetBuoyancy = CType(element.GetValue(BuoyancyProperty), Buoyancy)
    End Function
End Class
'</SnippetDOMain>

'<SnippetDOSimpleDP>
Public Class Fish
    Inherits Control

    Public Shared ReadOnly SpeciesProperty As DependencyProperty = _
    DependencyProperty.Register(
    "Species", _
    GetType(String), _
    GetType(Fish), _
    Nothing)
    Public Property Species As String
        Get
            Species = CType(GetValue(SpeciesProperty), String)
        End Get
        Set(value As String)
            SetValue(SpeciesProperty, value)
        End Set
    End Property
End Class
'</SnippetDOSimpleDP>
Public Class DependencyPropertyUtilities

    '<SnippetDOCheckClear>
    Public Shared Function ClearSetProperty(targetObject As DependencyObject, targetDP As DependencyProperty) As Boolean
        If targetObject Is Nothing Or targetDP Is Nothing Then
            Throw New ArgumentNullException()
        End If
        Dim localValue As Object = targetObject.ReadLocalValue(targetDP)
        If localValue = DependencyProperty.UnsetValue Then
            ClearSetProperty = False
        Else
            targetObject.ClearValue(targetDP)
            ClearSetProperty = True
        End If
    End Function
    '</SnippetDOCheckClear>
    '<SnippetDPGetMetadata>
    Public Shared Function GetDefaultValueForFrameworkDP(dpIdentifier As DependencyProperty) As Object
        Dim metadataInfo As PropertyMetadata = dpIdentifier.GetMetadata(GetType(FrameworkElement))
        GetDefaultValueForFrameworkDP = metadataInfo.DefaultValue
    End Function
    '</SnippetDPGetMetadata>
End Class